### ISSUE TITLE
Set Doxygen documentation for C code.

### ISSUE THREAD

**Issue by IshmaelBelghazi** [IshmaelBelghazi/bigoptim#6]

**Comment by tdhock** [comment]
documenting C code is technically not necessary for an R package, so I would advise to do this at the very end of GSOC only if you have extra time.

**Comment by IshmaelBelghazi** [comment]
Agreed. It is not essential at the moment. It may become more important later in the project when the C code gets more complex .
